### DISCUSSION TITLE
Change Script Kit Shortcut

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#531] [Docs]
<meta path="kit/change-main-shortcut">
      
# Change Script Kit Shortcut

Don't like `cmd+;`? Change it here!

**Comment by askarby** [comment]
This is an issue for me, due to the fact that we don't have a dedicated `;`-key on keyboards with danish layout (I assume this is the case for multiple keyboard layouts, at least "the nordic ones", eg. danish, swedish, norwegish etc. ).

It would be nice to have an option to reconfigure it?

Heck, I'll create a PR if you can point me in the direction of where the logic surrounding this key is located :)

**Reply by johnlindquist** [reply]
The option is under the "Kit" tab 😇

**Reply by askarby** [reply]
Thanks :)

**Comment by kenNash** [comment]
I'm sorry, I'm confused: what's the difference between `cmd+shift+;` and `cmd+;`?
How to change them both? (No `;` on my Italian keyboard.)

**Reply by johnlindquist** [reply]
@kenNash Script Kit no longer uses `cmd+shift+;` in the latest builds.

Here's what I mean by the "Kit" tab:
<img width="854" alt="CleanShot 2021-12-16 at 02 31 44@2x" src="https://user-images.githubusercontent.com/36073/146345409-e29b0d79-ae2f-423a-9dd3-b783541cf7f4.png">

The latest builds (releasing tomorrow) will also allow you to right-click on the icon to change the shortcut:
![CleanShot 2021-12-16 at 02 32 14@2x](https://user-images.githubusercontent.com/36073/146345535-8bac5c33-8c62-408d-9d8b-86d4ba4bfe47.png)

**Reply by kenNash** [reply]
Oh, thank you John!

The first one was the piece of information I was missing.

**Comment by prometheas** [comment]
Hey there! Although I'm able to remap the shortcut to stuff like cmd+shift+k, I seem to be unable to use cmd+space … I used to map that to Raycast, but I cannot seem to map this to ScriptKit.

And, for what it's worth, I've confirmed that Raycast is not indeed running, and that the chord doesn't map to any system-level shortcuts 😄

**Reply by johnlindquist** [reply]
Thanks for bumping this up again. I figured it out this morning* and I'll get it out in the next release.

P.S. - If you'd like to come hang in our new Discord server, we're over here: https://discord.gg/uyKBCGR2
I'm happy to help with script ideas and any other troubleshooting.

## *Bug explanation
Script Kit has a feature called "shortcodes" which allows you to assign a choice to a single key which you then trigger by hitting "space". Image assigning "A" to "apple", so you type "a", hit space, and it submits "Apple".

Relevant code: https://github.com/johnlindquist/kitapp/blob/main/src/components/input.tsx#L115

I had to disable this feature when in "Hotkey" mode (the mode that you see when assigning a shortcut) and now it works fine. Just a brain fart on my end forgetting that was hijacking the space bar.

**Reply by alkanna** [reply]
Hello @johnlindquist I seem to be hitting this issue again in the last beta release, is this supposed to be fixed since ?

**Reply by alkanna** [reply]
Setting this manyally in my .env works though :
```bash
KIT_MAIN_SHORTCUT="cmd space"
```

**Reply by johnlindquist** [reply]
@alkanna I'll investigate. I've been upgrading a lot of the dependencies, so maybe something impacted it. Thanks for calling it to my attention.
